katacomb/scripts/uv_export.py

```diff
@@ -1,104 +1,118 @@
-    parser.add_argument("--nvispio",
-                        default=1024,
-                        type=int,
-                        help="Number of visibilities "
-                             "read/written per IO call")
-    parser.add_argument("-ks", "--select",
-                        default="scans='track'; spw=0; corrprods='cross'",
-                        type=log_exception(log)(parse_python_assigns),
-                        help="katdal select statement "
-                             "Should only contain python "
-                             "assignment statements to python "
-                             "literals, separated by semi-colons")
-    parser.add_argument("--blavg",
-                        default=False, action="store_true",
-                        help="Apply baseline dependent averaging")
+                        default="INFO",
+                        help="Logging level. Default: %(default)s")
+    katdal_options(parser)
+    selection_options(parser)
+    export_options(parser)
     return parser
 
 
-setup_logging()
-
-args = create_parser().parse_args()
-
-KA = KatdalAdapter(katdal.open(args.katdata))
-
-with obit_context():
-    # Construct file object
-    aips_path = KA.aips_path(name=args.name, disk=args.disk,
-                            aclass=args.aclass, seq=args.seq, dtype="AIPS")
-
-    # Handle invalid sequence numbers
-    if args.seq is None or args.seq < 1:
-        aips_path.seq = next_seq_nr(aips_path)
-
-    # Apply the katdal selection
-    KA.select(**args.select)
-
-    # Fall over on empty selections
-    if not KA.size > 0:
-        raise ValueError("The katdal selection produced an empty dataset"
-                        "\n'%s'\n" % pretty(args.select))
-
-
-    # UV file location variables
-    with uv_factory(aips_path=aips_path, mode="w",
-                        nvispio=args.nvispio,
-                        table_cmds=KA.default_table_cmds(),
-                        desc=KA.uv_descriptor()) as uvf:
-
-        # Write history
-        uv_history_obs_description(KA, uvf)
-        uv_history_selection(args.select, uvf)
-
-        # Perform export to the file
-        uv_export(KA, uvf)
-
-    # Possibly perform baseline dependent averaging
-    if args.blavg == True:
-        task_kwargs = aips_path.task_input_kwargs()
-        task_kwargs.update(aips_path.task_output_kwargs(aclass='uvav'))
-        blavg = task_factory("UVBlAvg", **task_kwargs)
-
-        blavg.go()
+def main():
+    parser = create_parser()
+    args = parser.parse_args()
+    configure_logging(args)
+    log.info("Loading katdal dataset with applycal=%s", args.applycal)
+    katdata = katdal.open(args.katdata, applycal=args.applycal, **args.open_kwargs)
+
+    kat_select = setup_selection(katdata, args)
+    # Command line katdal selection overrides command line options
+    kat_select = recursive_merge(args.select, kat_select)
+
+    # UVBlAvg parameters
+    # TODO: Clean up this mess!
+    uvblavg_args = {}
```

Comment:
It seems to me that very similar code relating to the uvblavg_parm_file now resides in all three scripts (continuum_pipeline, continuum_image, uv_export), which obviously isn't ideal for future maintainability. Is it not possible to aggregate at least some of this into one function somewhere ? The same comment applies for the mfimage parameters in the two imaging scripts.

Reply:
okay, fair enough